crates/pathfinder/src/bin/pathfinder/main.rs

```diff
@@ -290,46 +292,100 @@ If you are trying to setup a custom StarkNet please use '--network custom',
-        result = sync_handle => {
-            match result {
-                Ok(task_result) => tracing::error!("Sync process ended unexpected with: {:?}", task_result),
-                Err(err) => tracing::error!("Sync process ended unexpected; failed to join task handle: {:?}", err),
+            // Monitor our spawned process tasks.
+            tokio::select! {
+                result = sync_handle => {
+                    match result {
+                        Ok(task_result) => tracing::error!("Sync process ended unexpected with: {:?}", task_result),
+                        Err(err) => tracing::error!("Sync process ended unexpected; failed to join task handle: {:?}", err),
+                    }
+                }
+                result = cairo_handle => {
+                    match result {
+                        Ok(task_result) => tracing::error!("Cairo process ended unexpected with: {:?}", task_result),
+                        Err(err) => tracing::error!("Cairo process ended unexpected; failed to join task handle: {:?}", err),
+                    }
+                }
+                _result = rpc_handle => {
+                    // This handle returns () so its not very useful.
+                    tracing::error!("RPC server process ended unexpected");
+                }
+                result = update_handle => {
+                    match result {
+                        Ok(_) => tracing::error!("Release monitoring process ended unexpectedly"),
+                        Err(err) => tracing::error!(error=%err, "Release monitoring process ended unexpectedly"),
+                    }
+                }
             }
         }
-        result = cairo_handle => {
-            match result {
-                Ok(task_result) => tracing::error!("Cairo process ended unexpected with: {:?}", task_result),
-                Err(err) => tracing::error!("Cairo process ended unexpected; failed to join task handle: {:?}", err),
+        ServerHandle::Ws(ws_server_handle) => {
+            let (rpc_handle, local_addr) = ws_server_handle.context("Starting the RPC server")?;
+            info!("📡 WS-RPC server started on: {}", local_addr);
+
+            let update_handle = tokio::spawn(update::poll_github_for_releases());
+
+            // We are now ready.
+            if let Some(ready) = pathfinder_ready {
+                ready.store(true, std::sync::atomic::Ordering::Relaxed);
             }
-        }
-        _result = rpc_handle => {
-            // This handle returns () so its not very useful.
-            tracing::error!("RPC server process ended unexpected");
-        }
-        result = update_handle => {
-            match result {
-                Ok(_) => tracing::error!("Release monitoring process ended unexpectedly"),
-                Err(err) => tracing::error!(error=%err, "Release monitoring process ended unexpectedly"),
+
+            // Monitor our spawned process tasks.
+            tokio::select! {
```

Comment:
This is the solution that is crappy but functional. We duplicate the whole block of tokio:select. We need to do that because httpServerHandle and wsServerHandle do not share any trait so we can't put either one or another in the same rpc_handle variable